packages/kbn-es-archiver/src/lib/docs/index_doc_records_stream.ts

```diff
@@ -44,7 +44,21 @@ export function createIndexDocRecordsStream(
       );
     });
 
-    const resp = await client.bulk({ body });
+    const maxTries = 5;
+    let resp;
+
+    for (let currentTry = 1; currentTry <= maxTries; currentTry++) {
+      try {
+        resp = await client.bulk({ body });
```

Comment:
Bulk usually doesn't throw an error when it fails, it instead reports failures in a 200 response because any one of the internal requests can fail. All the docs we send should include the id so I don't think there are any issues with this approach, but I just want to make sure it works.

Reply:
I'm not sure which timeout exactly I hit TBH. From what I've seen during my investigation, I suspect it is not coming from `bulk` directly, but I've tracked it down to this code line. Maybe it's a socket timeout or so.

Reply:
Do you have a error message or stack trace?

Reply:
I've just started my notebook again to reproduce and copy the stack trace, but it didn't happen any more via the regular CLI call. Earlier today, I hit the issue in about 2 of 5 tries or so, but now it didn't happen anymore in 20 tries. So it seems it was a local network flakiness.
With that suspicion I was able to reproduce the failure:
- Start some esArchive `load` process
- During the process (in my case, when the first `progress: n` was printed): disconnect the network of your notebook
- The process will run into a timeout:
```
 info [reporting/bwc/6_3] Loading "mappings.json"
 info [reporting/bwc/6_3] Loading "data.json.gz"
 info [reporting/bwc/6_3] Deleted existing index ".reporting-2018.05.06"
 info [reporting/bwc/6_3] Created index ".reporting-2018.05.06"
 info progress: 1
 info progress: 1
 info progress: 1
ERROR UNHANDLED ERROR
ERROR Error: Request Timeout after 30000ms
          at /home/robert/github/kibana/node_modules/elasticsearch/src/lib/transport.js:397:9
          at Timeout.<anonymous> (/home/robert/github/kibana/node_modules/elasticsearch/src/lib/transport.js:429:7)
          at listOnTimeout (internal/timers.js:554:17)
          at processTimers (internal/timers.js:497:7)
```
So the retry that comes in this PR will only help if the network connects again (and for long enough) before the last retry is over. That's still not ideal but could potentially help through some short lasting network problems.

Reply:
Yeah, this is a request timeout coming from the client itself because the overall request takes over 30 seconds. In this case the client will throw which will be "fixed" by this change. Only thing I don't like is that the long request time is probably caused by ES being busy, and resending the request probably won't help things and I think it's reasonable to expect that the subsequent requests will timeout too. I think instead we should try raising the timeout to 2 minutes or something. If the request takes longer than 2 minutes to complete then I think there's a real problem somewhere we should fix rather than retry the requests. Just specifying `requestTimeout: 2 * 60 * 1000` in the bulk params should do it.

Reply:
Sounds good. Gave it a local try and simulated a one minute network outage - it was able to finish the archive load process successfully. I will push the changes and adjust the PR title + description accordingly.

Reply:
Thanks for your help @spalger !